### PR TITLE
DOC/MAINT: single to double backticks to remove improper linking

### PR DESCRIPTION
**Pull request by Carreau**
This fixes a few obviously improper usage of single backticks where the content cannot be a target.

This is part of an effort to uniformize the usage of single and doube backtick in order to have proper linking to all objects

### CODE REVIEW

**Comment by Carreau** [comment]
cc @mdhaber

File: scipy/special/_basic.py [hunk 1]
```diff
@@ -3248,9 +3248,9 @@ def stirling2(N, K, *, exact=False):
         Temme for larger entries  of `N` and `K` that allows trading speed for
         accuracy. See [2]_ for a description. Temme approximation is used for
         values `n>50`. The max error from the DP has max relative error
-        `4.5*10^-16` for `n<=50` and the max error from the Temme approximation
-        has max relative error `5*10^-5` for `51 <= n < 70` and
-        `9*10^-6` for `70 <= n < 101`. Note that these max relative errors will
+        ``4.5*10^-16`` for `n<=50` and the max error from the Temme approximation
```
Comment (mdhaber) [inline scipy/special/_basic.py:3251]:
```suggestion
        ``4.5*10^-16`` for ``n<=50`` and the max error from the Temme approximation
```

**Review by mdhaber — APPROVED** [review]
After a quick pass, LGTM except for the one additional correction I noticed. If CI passes, I'll commit and merge tomorrow unless someone beats me to it.

File: scipy/special/_add_newdocs.py [hunk 2]
```diff
@@ -5582,10 +5582,10 @@ def add_newdoc(name, doc):
     Parameters
     ----------
     a : array_like
-        `a` parameter values of `gdtr(a, b, x)`. `1/a` is the "scale"
+        `a` parameter values of ``gdtr(a, b, x)`. ```1/a`` is the "scale"
```
Comment (j-bowhay) [inline scipy/special/_add_newdocs.py:5585]:
```suggestion
        `a` parameter values of ``gdtr(a, b, x)`. ``1/a`` is the "scale"
```

**Comment by Carreau** [comment]
Thanks for the extra find and the stray backtick

File: scipy/special/_add_newdocs.py [hunk 3]
```diff
@@ -5582,10 +5582,10 @@ def add_newdoc(name, doc):
     Parameters
     ----------
     a : array_like
-        `a` parameter values of `gdtr(a, b, x)`. `1/a` is the "scale"
+        `a` parameter values of ``gdtr(a, b, x)`. ``1/a`` is the "scale"
```
Comment (j-bowhay) [inline scipy/special/_add_newdocs.py:5585]:
```suggestion
        `a` parameter values of ``gdtr(a, b, x)``. ``1/a`` is the "scale"
```
```suggestion
        `a` parameter values of ``gdtr(a, b, x)`. ``1/a`` is the "scale"
```

File: scipy/special/_add_newdocs.py [hunk 4]
```diff
@@ -5582,10 +5582,10 @@ def add_newdoc(name, doc):
     Parameters
     ----------
     a : array_like
-        `a` parameter values of `gdtr(a, b, x)`. `1/a` is the "scale"
+        `a` parameter values of ``gdtr(a, b, x)``. ``1/a`` is the "scale"
         parameter of the gamma distribution.
     b : array_like
-        `b` parameter values of `gdtr(a, b, x)`. `b` is the "shape" parameter
+        `b` parameter values of ``gdtr(a, b, x)`. `b` is the "shape" parameter
```
Comment (j-bowhay) [inline scipy/special/_add_newdocs.py:5588]:
```suggestion
        `b` parameter values of ``gdtr(a, b, x)``. `b` is the "shape" parameter
```

**Review by jarrodmillman — APPROVED** [review]

**Review by lucascolley — APPROVED** [review]
thanks all!

**Pull request merged**